src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,174 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.NoSuchElementException;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            Node<T> node = new Node<>(null, value, null);
+            head = node;
+            tail = node;
+        } else {
+            Node<T> node = new Node<>(tail, value, tail.next);
+            tail.next = node;
+            tail = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds for list size "
+                    + size);
+        }
+        if (index == size) {
+            add(value);
+        } else if (index == 0) {
+            Node<T> node = new Node<>(head.prev, value, head);
+            head.prev = node;
+            head = node;
+            size++;
+        } else {
+            Node<T> current = head;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+            Node<T> node = new Node<>(current.prev, value, current);
+            current.prev.next = node;
+            current.prev = node;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        if (index == 0) {
+            return head.value;
+        }
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        return current.value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        checkElement(index);
+        T result;
+        if (index == 0) {
+            result = head.value;
+            head.value = value;
+            return result;
+        }
+        if (index == size - 1) {
+            result = tail.value;
+            tail.value = value;
+            return result;
+        }
+        Node<T> current = head;
+        for (int i = 0; i < index; i++) {
+            current = current.next;
+        }
+        result = current.value;
+        current.value = value;
+        return result;
```

Comment:
too big method, let's rewrite your logic:
1. Find node by index
2. Create variable for storing current value of node 
3. Update value for node
4. Return old value of node